### PR TITLE
Add a minimal s03 example

### PR DESCRIPTION
**Pull request by MikeG112**
I added a minimal example of training a model of the type we described on MNIST data.

### CODE REVIEW

**Comment by MikeG112** [comment]
It seems to train decently well (though I experienced some gradient clipping on a few runs), and on my machine running it for 3 epochs with a learning rate of .01 resulted in average ROC AUC of .99 and accuracy of 93%, which is better than I expected for just a first pass. Didn't do any work on the underlying `euclidean_model` here, just used a very basic combination of a few convolutional layers and linear layers pulled from a random MNIST example. I should try running _just_ that model to investigate whether the whole S03 aspect was even that relevant, or just made a decent model slightly worse lol, but haven't yet. One cool thing about this, in addition to the performance being not bad for a relatively small model, is that by design of the `SO3Block` the model outputs for an image factor through embeddings of images into $\mathbb{R}^{3}$. it would be really cool if these embeddings were of a different nature than embeddings you might get by e.g. just taking the euclidean backbone and having it factor through a second to last linear layer of size 3. Gonna try investigating the embeddings a bit more next, and try a few other types of training examples to see how generalization goes.
